pep-9999.rst

```diff
@@ -175,8 +190,33 @@ performed):
                 pep508 += " @ " + vcs + "+" requirement[vcs]
                 if "revision" in requirement:
                     pep508 += "@" + revision
+        markers = ""
+        for variable in (
+            "os_name",
+            "sys_platform",
+            "platform_machine",
+            "platform_python_implementation",
+            "platform_release",
+            "platform_system",
+            "platform_version",
+            "python_version",
+            "python_full_version",
+            "implementation_name",
+            "implementation_version",
+            "extra",
+        ):
+            key = variable.replace("_", "-")
```

Comment:
Would this be better? `variable.replace("_", ".", 1)`?

Reply:
No, as a key with a dot in it is not a dotted-key. Example:
```toml
"this.is.a.key.with.dots" = "spam"
dotted.key = "eggs"
```
```json
{"this.is.a.key.with.dots": "spam", "dotted": {"key": "eggs"}}
```
For dotted-key notation, you'll need to do JSON path parseing (`jquery`, `jmespath`, `jsonpath`, etc)

Reply:
The reason hyphens are used is to match the style in the rest of PEP 621. I think it's a useful transformation, as the marker variable names must be valid Python names with no hyphens

Reply:
But that is what I wanted. :)

`"python": { "version": "", "full_version": ""}`

Reply:
The other option is to build a custom map from requirement table keys to environment marker variables, which is cumbersome

Reply:
As in, it would become:
```python
if "name" in requirement.get("os", {}):
    markers.append("os_name " + requirement["os"]["name"])
...
```

Reply:
Full example:
```python
markers = []
for variable, key1, key2 in (
    ("os_name", "os", "name"),
    ("sys_platform", "sys", "platform"),
    ("platform_machine", "platform", "machine"),
    (
        "platform_python_implementation",
        "platform",
        "python-implementation",
    ),
    ("platform_release", "platform", "release"),
    ("platform_system", "platform", "system"),
    ("platform_version", "platform", "version"),
    ("python_version", "python", "version"),
    ("python_full_version", "python", "full-version"),
    ("implementation_name", "implementation", "name"),
    ("implementation_version", "implementation", "version"),
):
    if key2 in requirement.get(key1, {}):
        markers.append(variable + " " + requirement[key1][key2])
if "extra" in requirement:
    markers.append(requirement["extra"])
if "markers" in requirement:
    markers_explicit = requirement["markers"]
    if markers:
        markers_explicit = "(" + markers_explicit + ")"
    markers.append(markers_explicit)
if markers:
    pep508 += " ; " + " and ".join(markers)
```

Reply:
And if that is to be accepted, would you show the example with `python.version = "< '2.7'"` or `python = { version = "< '2.7'" }`?

Reply:
I was thinking the former, as you have added in #6. Although, I do like the simplicitly of your current proposal ("just replace the _ with -").

Reply:
The only thing my suggestion really allows for is the flexibility to define these as their own tables. Which I think is unnecessary.

Reply:
It would be adding grouping and hierarchy for not many child items. Perhaps some markers could be made in to sub-tables while others are left as top-level strings? That would be inconsistent.

Reply:
I vote lets keep it consistent for now.

Reply:
I'll see if #6 gets any traction